engine/src/main/java/io/camunda/zeebe/engine/processing/bpmn/container/MultiInstanceBodyProcessor.java

```diff
@@ -432,4 +496,66 @@ private DirectBuffer insertAt(
     }
     return Either.right(false);
   }
+
+  private DirectBuffer wrapNumberOfElementInstances(
+      final MutableDirectBuffer variableBuffer,
+      final DirectBuffer variableView,
+      final long numberOfElementInstances) {
+    variableWriter.wrap(variableBuffer, 0);
+
+    variableWriter.writeInteger(numberOfElementInstances);
+    final var length = variableWriter.getOffset();
+
+    variableView.wrap(variableBuffer, 0, length);
+    return variableView;
+  }
```

Comment:
🔧  I think we can merge this method with `wrapLoopCounter` into something like `wrapVariable` or so, because these do the same work